src/P2P/MulticastBroadcaster.ts

```diff
@@ -0,0 +1,193 @@
+
+import PeerStore from '@polykey/PeerStore/PeerStore'
+import KeyManager from '@polykey/KeyManager'
+import RPCMessage from '@polykey/RPC/RPCMessage'
+import dgram from 'dgram'
+import crypto from 'crypto'
+import { EventEmitter } from 'events'
+
+// This module is based heavily on libp2p's mDNS module:
+// https://github.com/libp2p/js-libp2p-mdns
+// It is supposed to discover peers on the local network
+// This module was also generated with the help of:
+// https://nrempel.com/using-udp-multicast-with-node-js/
+//
+// """
+// In computer networking, the multicast DNS (mDNS) protocol
+// resolves hostnames to IP addresses within small networks
+// that do not include a local name server
+// """
+
+const UDP_MULTICAST_PORT = 5353;
+const UDP_MULTICAST_ADDR = "224.0.0.251";
+
+type PeerMessage = {
+  encryptedLocalPubKey: Buffer
+  encryptedPeerPubKey: Buffer
+  rawRandomMessage: Buffer
+  encryptedRandomMessage: Buffer
+}
+
+class MulticastBroadcaster extends EventEmitter {
+  peerStore: PeerStore
+  keyManager: KeyManager
+
+  socket: dgram.Socket
+
+  interval: number
+  queryInterval: NodeJS.Timeout | null
+  peerPubKeyMessages: Map<string, PeerMessage> = new Map()
+  constructor(
+    peerStore: PeerStore,
+    keyManager: KeyManager
+  ) {
+    super()
+
+    this.peerStore = peerStore
+    this.keyManager = keyManager
+
+    this.interval = (1e3)
+    this.queryInterval = null
+  }
+
+
+  queryLAN() {
+    const query = () => {
+      for (const pubKey of this.peerPubKeyMessages.keys()) {
+        const peerMessage = this.peerPubKeyMessages.get(pubKey)
+        if (peerMessage) {
+          const handshakeMessage = RPCMessage.encodeHandshakeMessage(
+            peerMessage.encryptedPeerPubKey,
+            peerMessage.encryptedLocalPubKey,
+            peerMessage.encryptedRandomMessage
+          )
+
+          this.socket.send(handshakeMessage, 0, handshakeMessage.length, UDP_MULTICAST_PORT, UDP_MULTICAST_ADDR, () => {
+            console.info(`Sending message to peer`);
+          });
+        }
+
+      }
+    }
+
+    // Immediately start a query, then do it every interval.
+    query()
+    return setInterval(query, this.interval)
+  }
+
+  /**
+   * Start sending queries to the LAN.
+   *
+   * @returns {void}
+   */
+  async start(): Promise<void> {
```

Comment:
Please keep return types consistent. I think some of your methods have no return type. If we specify return types, specify it for all.

Reply:
I have made all return types explicit :+1: